src/main/java/backend/control/RepoOpControl.java

```diff
@@ -42,6 +64,26 @@ public RepoOpControl(RepoIO repoIO) {
         return result;
     }
 
+    /**
+     * Updates repository stored locally with data from a GitHubModelUpdatesData object.
+     * Set syncOperation to queue this operation in the blocking queue for the updating repository
+     * @param updates
+     * @param syncOperation
+     * @return
+     */
+    public CompletableFuture<Model> updateLocalModel(GitHubModelUpdatesData updates,
+                                                     boolean syncOperation) {
+        init(updates.getRepoId());
+        CompletableFuture<Model> result = new CompletableFuture<>();
+        UpdateLocalModelOp op = new UpdateLocalModelOp(models, updates, result);
+        if (syncOperation) {
```

Comment:
Hmm, this feels a bit weird. If we don't need to synchronise it, we can just call the Logic version of this?